### PR TITLE
[neuralnet] Add null character when load optizmier type from bin file

### PR DESCRIPTION
**Pull request by lhs8928**
- Added missing null character when load optimizer type from bin file

Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

File: nntrainer/models/neuralnet.cpp [hunk 1]
```diff
@@ -382,7 +382,8 @@ void NeuralNetwork::load(const std::string &file_path,
       /// this is assuming that the failure is allowed at the end of the file
       /// read. so, after this line, additional read shouldn't be called
       if (opt && istrequal(opt->getType(), "adam")) {
-        char opt_type[4];
+        char opt_type[5];
+        opt_type[4] = '\0';
         model_file.read(opt_type, 4);
```
Comment (songgot) [inline nntrainer/models/neuralnet.cpp:387]: How about check the return value if 0 or ssize_t ?

File: nntrainer/models/neuralnet.cpp [hunk 2]
```diff
@@ -382,8 +382,9 @@ void NeuralNetwork::load(const std::string &file_path,
       /// this is assuming that the failure is allowed at the end of the file
       /// read. so, after this line, additional read shouldn't be called
       if (opt && istrequal(opt->getType(), "adam")) {
-        char opt_type[4];
-        model_file.read(opt_type, 4);
+        std::string opt_type;
```
Comment (jijoongmoon) [inline nntrainer/models/neuralnet.cpp:385]: you might need to change the save function as well.

Comment (lhs8928) [reply nntrainer/models/neuralnet.cpp:386]: Done

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**